### ISSUE TITLE
install instructions

### ISSUE THREAD

**Issue by ImaginaryDevelopment** [jmecosta/sonar-fsharp-plugin#12]
how do I install this into sonar server? 
I see that it isn't in the list of sonar's update center. do I have to download the source and build it? how does someone install a sonar-plugin manually?

( also do I need a special runner, or will the one that was downloaded that seems to work fine for C# work?)

**Comment by jmecosta** [comment]
ive updated the readme just now, you should not require anything other than the installation of the plugin in server. everything is setup during analysis.

**Comment by ImaginaryDevelopment** [comment]
If that's all that is needed, I'm not sure why this is happening, any thoughts?

Failed: System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Build.Framework, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

**Comment by ImaginaryDevelopment** [comment]
that was using the deprecated for msbuild runner. this one however is the latest and has the same initial error

```
16:23:15.370 INFO  - Sensor org.sonar.plugins.fsharp.FSharpSensor
16:23:16.086 INFO  - [|"/i:C:\TFS\PracticeManagement\dev\PracticeManagement\.sonarqube\out\.sonar\PM_PM_8CF8836A-5AAD-4D56-8BC0-D1522066B027\fs-analysis-input.xml";
16:23:16.086 INFO  -   "/o:C:\TFS\PracticeManagement\dev\PracticeManagement\.sonarqube\out\.sonar\PM_PM_8CF8836A-5AAD-4D56-8BC0-D1522066B027\fs-analysis-output.xml"|]
16:23:16.393 INFO  -     Failed: System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Build.Framework, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
16:23:16.393 INFO  - File name: 'Microsoft.Build.Framework, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.MSBuildResolver.ResolveCore(ResolutionEnvironment resolutionEnvironment, Tuple`2[] references, String targetFrameworkVersion, FSharpList`1 targetFrameworkDirectories, String targetProcessorArchitecture, String outputDirectory, String fsharpCoreExplicitDirOrFSharpBinariesDir, FSharpList`1 explicitIncludeDirs, String implicitIncludeDir, String frameworkRegistryBase, String assemblyFoldersSuffix, String assemblyFoldersConditions, Boolean allowRawFileName, FSharpFunc`2 logmessage, FSharpFunc`2 logwarning, FSharpFunc`2 logerror)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.MSBuildResolver.CallResolveCore@423(String targetProcessorArchitecture, String targetFrameworkVersion, FSharpList`1 targetFrameworkDirectories, ResolutionEnvironment resolutionEnvironment, String outputDirectory, FSharpFunc`2 logwarning, FSharpFunc`2 logmessage, FSharpFunc`2 logerror, String implicitIncludeDir, String fsharpCoreExplicitDirOrFSharpBinariesDir, String frameworkRegistryBase, FSharpList`1 explicitIncludeDirs, String assemblyFoldersSuffix, String assemblyFoldersConditions, Tuple`2[] references, Boolean allowRawFileName)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.MSBuildResolver.Resolve(ResolutionEnvironment resolutionEnvironment, Tuple`2[] references, String targetFrameworkVersion, FSharpList`1 targetFrameworkDirectories, String targetProcessorArchitecture, String outputDirectory, String fsharpCoreExplicitDirOrFSharpBinariesDir, FSharpList`1 explicitIncludeDirs, String implicitIncludeDir, String frameworkRegistryBase, String assemblyFoldersSuffix, String assemblyFoldersConditions, FSharpFunc`2 logmessage, FSharpFunc`2 logwarning, FSharpFunc`2 logerror)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.Build.Resolve@3018-1(TcConfig tcConfig, ResolveAssemblyReferenceMode mode, range errorAndWarningRange, String targetFrameworkMajorMinor, String targetProcessorArchitecture, String outputDirectory, FSharpList`1 targetFrameworkDirectories, Tuple`2[] references, Boolean showMessages)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.Build.TcConfig.TryResolveLibsUsingMSBuildRules(TcConfig tcConfig, FSharpList`1 originalReferences, range errorAndWarningRange, ResolveAssemblyReferenceMode mode)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.Build.TcAssemblyResolutions.Resolve(TcConfig tcConfig, FSharpList`1 assemblyList, FSharpList`1 knownUnresolved)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.Build.GetAssemblyResolutionInformation(TcConfig tcConfig)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.Build.ScriptPreprocessClosure.GetFullClosureOfScriptSource(String filename, String source, CodeContext codeContext, Boolean useMonoResolution, Boolean useFsiAuxLib, LexResourceManager lexResourceManager, FSharpFunc`2 applyCommmandLineArgs)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.Build.LoadClosure.ComputeClosureOfSourceText(String filename, String source, CodeContext implicitDefines, Boolean useMonoResolution, Boolean useFsiAuxLib, LexResourceManager lexResourceManager, FSharpFunERROR: Error during Sonar runner execution
c`2 applyCompilerOptions)
16:23:16.393 INFO  -    at <StartupCode$FSharp-Compiler-Service>.$Service.GetProjectOptionsFromScript@2435.Invoke(Unit unitVar0)
org.sonar.runner.impl.RunnerException: Unable to execute Sonar
        at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:91)
16:23:16.393 INFO  -    at Microsoft.FSharp.Compiler.SourceCodeServices.Reactor.EnqueueAndAwaitOpAsync@203.Invoke(Unit unitVar0)
        at org.sonar.runner.impl.BatchLauncher$1.run(BatchLauncher.java:75)
16:23:16.393 INFO  - --- End of stack trace from previous location where exception was thrown ---
        at java.security.AccessController.doPrivileged(Native Method)
16:23:16.393 INFO  -    at Microsoft.FSharp.Control.AsyncBuilderImpl.commit[a](Result`1 res)
        at org.sonar.runner.impl.BatchLauncher.doExecute(BatchLauncher.java:69)
        at org.sonar.runner.impl.BatchLauncher.execute(BatchLauncher.java:50)
16:23:16.393 INFO  -    at Microsoft.FSharp.Control.CancellationTokenOps.RunSynchronously[a](CancellationToken token, FSharpAsync`1 computation, FSharpOption`1 timeout)
        at org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:102)
16:23:16.393 INFO  -    at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken)
16:23:16.393 INFO  -    at FsSonarRunnerCore.SQAnalyser.GatherMetrics(String path, String input, SonarResoureMetrics resourceMetric)
16:23:16.393 INFO  -    at FsSonarRunnerCore.SQAnalyser.GatherMetricsForResource(String path, String input)
16:23:16.393 INFO  -    at FsSonarRunnerCore.SQAnalyser.RunAnalyses(String path, String input, String inputXmlConfig)
16:23:16.393 INFO  -    at Program.main@49-1.Invoke(String c)
        at org.sonar.runner.api.Runner.execute(Runner.java:100)
16:23:16.393 INFO  -    at Microsoft.FSharp.Collections.SeqModule.Iterate[T](FSharpFunc`2 action, IEnumerable`1 source)
16:23:16.393 INFO  -    at Program.main$cont@37(String[] argv, FSharpMap`2 arguments, Unit unitVar)
16:23:16.393 INFO  -
16:23:16.393 INFO  - WRN: Assembly binding logging is turned OFF.
        at org.sonar.runner.Main.executeTask(Main.java:70)
        at org.sonar.runner.Main.execute(Main.java:59)
        at org.sonar.runner.Main.main(Main.java:53)
16:23:16.393 INFO  - To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
16:23:16.393 INFO  - Note: There is some performance penalty associated with assembly bind failure logging.
16:23:16.393 INFO  - To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
Total time: 2.797s
Final Memory: 6M/121M
INFO: ------------------------------------------------------------------------
16:23:16.501  Creating a summary markdown file...
Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: C:\TFS\PracticeManagement\dev\PracticeManagement\.sonarqube\out\.sonar\PM_PM_8CF8836A-5AAD-4D56-8BC0-D1522066B027\fs-analysis-output.xml (The system cannot find the file specified)
        at com.google.common.base.Throwables.propagate(Throwables.java:156)
        at org.sonar.plugins.fsharp.FSharpSensor$AnalysisResultImporter.parse(FSharpSensor.java:234)
        at org.sonar.plugins.fsharp.FSharpSensor.importResults(FSharpSensor.java:195)
        at org.sonar.plugins.fsharp.FSharpSensor.analyse(FSharpSensor.java:105)
        at org.sonar.batch.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:58)
        at org.sonar.batch.phases.SensorsExecutor.execute(SensorsExecutor.java:50)
        at org.sonar.batch.phases.PhaseExecutor.
execute(PhaseExecutor.java:98)
        at org.sonar.batch.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:185)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:132)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:117)
        at org.sonar.batch.scan.ProjectScanContainer.scan(ProjectScanContainer.java:243)
        at org.sonar.batch.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:238)
        at org.sonar.batch.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:236)
        at org.sonar.batch.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:228)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:132)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:117)
        at org.sonar.batch.task.ScanTask.execute(ScanTask.java:55)
        at org.sonar.batch.task.TaskContainer.doAfterStart(TaskContainer.java:86)
        at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:132)
        at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:117)
        at org.sonar.batch.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:122)
        at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:119)
        at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:79)

```

**Comment by jmecosta** [comment]
Do you have vs 2013 installed?

**Comment by jmecosta** [comment]
If not I might need to release a new version that does not have dependency
to this dll but to the 2015 versionn

**Comment by ImaginaryDevelopment** [comment]
I do not have 2013 installed. but it should be possible to just add a
binding redirect rather to the config files than releasing a different
version. at least I would think.

**Comment by jmecosta** [comment]
is not that easy to do that since the exe is embembed in the jar and the all folder when is extracted will overwrite the current app.config. 

but you can install this
https://www.microsoft.com/en-us/download/details.aspx?id=40760

it should be ok in your case

**Comment by ImaginaryDevelopment** [comment]
Thanks, I'm going to try that

**Comment by ImaginaryDevelopment** [comment]
worked, I'd love you to consider publishing different versions for different dependencies, but with sonar saying [itself](http://docs.sonarqube.org/display/SONARNEXT/Analyzing+with+SonarQube+Scanner+for+MSBuild) that they have deprecated pre msbuild 14 support.. maybe just the new .jar ?

**Comment by jmecosta** [comment]
In this case it really does not matter.  The only thing needed is to
include the missing dlls in the zip and should be OK.

**Comment by ImaginaryDevelopment** [comment]
should I leave this issue open until you've done that, or close it now?

**Comment by jmecosta** [comment]
Leave open so I don't forget

**Comment by jmecosta** [comment]
@ImaginaryDevelopment let me know if RC3 handles your issue

**Comment by milbrandt** [comment]
@ImaginaryDevelopment Installation instruction in PR #41 and some other rework was done. Could you please test if with v1.0.5 (coming soon) your issues are solved?

**Comment by milbrandt** [comment]
No reply for 3 weeks, install instructions in `Readme.md` are updated, new release which should no longer rely on MSBuild 12 is going to be released with 1.1.0 (using .NET 4.7.2 instead of .NET 4.5).
